edweiss-app/app/(app)/quiz/createQuizPage.tsx

```diff
@@ -0,0 +1,612 @@
+				<TTouchableOpacity flex={1} radius={"xl"} p={"sm"} ml='sm' backgroundColor='red' flexDirection='row' flexColumnGap='sm' onPress={() => props.removeExerciseFromList(props.index)}>
+					<Icon name='trash-bin-sharp' size='xl' mt='md' ml='sm' />
+
+					<TText color='text' ml='lg' mt='md' size='lg'>
+						Delete
+					</TText>
+				</TTouchableOpacity>
+				<TTouchableOpacity flex={1} radius={"xl"} p={"sm"} mr='sm' backgroundColor='blue' flexDirection='row' flexColumnGap='sm' onPress={() => {
+					console.log("in edit button : " + props.exercise.question); console.log("index : " + props.index); props.secondModalRef.current?.present()
+				}}>
+					<Icon name='pencil-sharp' size='xl' mt='md' />
+					<TText color='text' size='lg' ml='lg'>
+						Edit
+					</TText>
+				</TTouchableOpacity>
+
+			</TView>
+
+			<EditSecondModal modalRef={props.secondModalRef} editExercise={props.editExercise} exercise={props.exercise} index={props.index}></EditSecondModal>
+
+		</ModalContainerScrollView>
+	</>
+
+
+	);
+
+};
+
+export const EditSecondModal: ReactComponent<{ modalRef: RefObject<BottomSheetModalMethods>, editExercise: (exercise: Quizzes.Exercise, index: number) => void, exercise: Quizzes.Exercise, index: number }> = (props) => {
+	const previousQuestion = props.exercise.question
+	if (props.exercise.type == "MCQ") {
+		const previousPropositions: [string, boolean][] = [];
+		const answers = props.exercise.answersIndices
+		props.exercise.propositions.forEach((proposition, index) => {
+			previousPropositions.push([proposition.description, answers.find(element => element == index) != undefined])
+		})
+		return (
+			<ModalContainerScrollView modalRef={props.modalRef}>
+				<TScrollView>
+					<MCQFields modalRef={props.modalRef} editExercise={props.editExercise} previousQuestion={previousQuestion} previousPropositions={previousPropositions} index={props.index}></MCQFields>
+				</TScrollView>
+			</ModalContainerScrollView>
+
+		);
+	}
+	else {
+
+		return (
+			<ModalContainerScrollView modalRef={props.modalRef}>
+				<TScrollView>
+					<TFFields modalRef={props.modalRef} previousQuestion={previousQuestion} previousBoolean={props.exercise.answer} editExercise={props.editExercise} index={props.index}></TFFields>
+				</TScrollView>
+			</ModalContainerScrollView>
+
+		);
+	}
+
+
+};
+export const MCQFields: ReactComponent<{ addToExerciseList?: (exercise: Quizzes.Exercise) => void, editExercise?: (exercise: Quizzes.Exercise, index: number) => void, modalRef: RefObject<BottomSheetModalMethods>, previousPropositions?: [string, boolean][], previousQuestion?: string, index?: number }> = (props) => {
+	const [question, setQuestion] = useState<string>(props.previousQuestion == undefined ? "" : props.previousQuestion);
+	//const [answer, setAnswer] = useState<number[]>([]);
+	const [propositions, setPropositions] = useState<[string, boolean][]>(props.previousPropositions == undefined ? [] : props.previousPropositions);
```

Comment:
You can define specific types for propositions and exercises, by exemple : 
```

type Proposition = { description: string; isCorrect: boolean };
type Exercise = {
    question: string;
    type: "MCQ" | "TF";
    propositions?: Proposition[];
    answersIndices?: number[];
    // ...
};

```

Reply:
If you are talking about the model for quizzes, I thought about changing the types; but the code is already there and changing the model would add a lot of refactoring. That being said, you are absolutely right that this is a clearer approach.